crates/oxc_linter/src/rules/jest/no_jasmine_globals.rs

```diff
@@ -0,0 +1,276 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::Reference;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, Fix};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(jest/no-jasmine-globals): {0:?}")]
+#[diagnostic(severity(warning), help("{1:?}"))]
+struct NoJasmineGlobalsDiagnostic(pub &'static str, pub &'static str, #[label] pub Span);
+
+/// <https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-jasmine-globals.md>
+#[derive(Debug, Default, Clone)]
+pub struct NoJasmineGlobals;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// This rule reports on any usage of Jasmine globals, which is not ported to Jest, and suggests alternatives from Jest's own API.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// jasmine.DEFAULT_TIMEOUT_INTERVAL = 5000;
+    /// test('my test', () => {
+    ///     pending();
+    /// });
+    /// test('my test', () => {
+    ///     jasmine.createSpy();
+    /// });
+    /// ```
+    NoJasmineGlobals,
+    restriction
+);
+
+const JASMINE_NAMES: [&str; 5] = ["spyOn", "spyOnProperty", "fail", "pending", "jasmine"];
+
+impl Rule for NoJasmineGlobals {
+    fn run_once(&self, ctx: &LintContext) {
+        let symbol_table = ctx.symbols();
+        let jasmine_references = ctx
+            .scopes()
+            .root_unresolved_references()
+            .iter()
+            .filter(|(key, _)| JASMINE_NAMES.contains(&key.as_str()));
+
+        for (name, reference_ids) in jasmine_references {
+            for &reference_id in reference_ids {
+                let reference = symbol_table.get_reference(reference_id);
+                if let Some((error, help)) = get_non_jasmine_property_messages(name) {
+                    ctx.diagnostic(NoJasmineGlobalsDiagnostic(error, help, reference.span()));
+                } else {
+                    diagnostic_jasmine_property(reference, ctx);
+                }
+            }
+        }
+    }
+}
+
+fn diagnostic_jasmine_property(reference: &Reference, ctx: &LintContext) {
+    let parent = ctx.nodes().parent_node(reference.node_id());
+    if let Some(parent) = parent {
+        if let AstKind::MemberExpression(member_expr) = parent.kind() {
+            let Some(property_name) = member_expr.static_property_name() else { return };
+
+            if property_name == "DEFAULT_TIMEOUT_INTERVAL" {
+                if let Some(ancestor) = ctx
```

Comment:
I want to find the AssignmentExpression from `jasmine. DEFAULT_TIMEOUT_INTERVAL = 5`, is there some better way?🤔

Reply:
Maybe use a plain `run` method? Going backwards from an identifier reference feels a bit strange.